Failing tests when importing json

I just noticed that there are currently two failing tests when importing json instead of ujson (related #99):
```
====================================================================================== FAILURES ======================================================================================
_____________________________________________________________________________________ test_data ______________________________________________________________________________________

    def test_data():
        response = client.post('http://example.com/data/', json={"hello": 123})
        assert response.json() == {'data': {"hello": 123}}
    
        response = client.post('http://example.com/data/')
        assert response.json() == {'data': None}
    
        response = client.post('http://example.com/data/', data={'abc': 123})
>       assert response.json() == {'data': {'abc': ['123']}}
E       AssertionError: assert {'data': {'abc': '123'}} == {'data': {'abc': ['123']}}
E         Differing items:
E         {'data': {'abc': '123'}} != {'data': {'abc': ['123']}}
E         Use -v to get the full diff

tests/test_http.py:198: AssertionError
_____________________________________________________________________________________ test_field _____________________________________________________________________________________

    def test_field():
        response = client.post('http://example.com/field/', json={"field": 123})
        assert response.json() == {'field': 123}
    
        response = client.post('http://example.com/field/', data={'field': 123})
        assert response.json() == {'field': '123'}
    
        csv_file = ('report.csv', '1,2,3\n4,5,6\n')
>       response = client.post('http://example.com/field/', files={'field': csv_file})

tests/test_http.py:220: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/lib/python3.6/site-packages/requests/sessions.py:535: in post
    return self.request('POST', url, data=data, json=json, **kwargs)
apistar/test.py:102: in request
    return super().request(method, url, **kwargs)
venv/lib/python3.6/site-packages/requests/sessions.py:488: in request
    resp = self.send(prep, **send_kwargs)
venv/lib/python3.6/site-packages/requests/sessions.py:609: in send
    r = adapter.send(request, **kwargs)
apistar/test.py:72: in send
    wsgi_response = self.wsgi_app(environ, start_response)
apistar/app.py:101: in func
    pipelines.run_pipeline(app.router.exception_pipeline, state)
apistar/pipelines.py:62: in run_pipeline
    state[output] = function(**kwargs)
apistar/app.py:98: in func
    state[output] = function(**kwargs)
tests/test_http.py:58: in get_field
    return http.Response({'field': field})
apistar/http.py:246: in __init__
    content = json.dumps(data).encode('utf-8')
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py:231: in dumps
    return _default_encoder.encode(obj)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py:257: in iterencode
    return _iterencode(o, 0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.encoder.JSONEncoder object at 0x101fda908>, o = <FileStorage: 'report.csv' (None)>

    def default(self, o):
        """Implement this method in a subclass such that it returns
            a serializable object for ``o``, or calls the base implementation
            (to raise a ``TypeError``).
    
            For example, to support arbitrary iterators, you could
            implement default like this::
    
                def default(self, o):
                    try:
                        iterable = iter(o)
                    except TypeError:
                        pass
                    else:
                        return list(iterable)
                    # Let the base class default method raise the TypeError
                    return JSONEncoder.default(self, o)
    
            """
        raise TypeError("Object of type '%s' is not JSON serializable" %
>                       o.__class__.__name__)
E       TypeError: Object of type 'FileStorage' is not JSON serializable

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/encoder.py:180: TypeError
======================================================================== 2 failed, 96 passed in 2.65 seconds =========================================================================
```

--- a/README.md
+++ b/README.md
@@ -655,6 +655,8 @@ A pull request [has been issued](https://github.com/TechEmpower/FrameworkBenchma
 to add API Star to future rounds of the TechEmpower benchmarks. In the future we
 plan to be adding more realistic & useful test types, such as database query performance.
 
+API Star optionally supports the `ujson` package for improvements in serialization performance. Currently `ujson` will automatically be used if the package is installed.
+
 ---
 
 # Deployment

--- a/apistar/__init__.py
+++ b/apistar/__init__.py
@@ -11,7 +11,7 @@ from apistar.routing import Route
 from apistar.templating import Template, Templates
 from apistar.test import TestClient
 
-__version__ = '0.1.14'
+__version__ = '0.1.15'
 __all__ = [
     'App', 'Route', 'Request', 'Response', 'Template', 'Templates', 'TestClient'
 ]

--- a/requirements.txt
+++ b/requirements.txt
@@ -4,7 +4,6 @@ coreapi
 jinja2
 pytest
 requests
-ujson
 werkzeug
 
 # Optional

--- a/setup.py
+++ b/setup.py
@@ -62,7 +62,6 @@ setup(
         'jinja2',
         'pytest',
         'requests',
-        'ujson',
         'werkzeug'
     ],
     classifiers=[

--- a/tests/test_http.py
+++ b/tests/test_http.py
@@ -2,6 +2,21 @@ from apistar import App, Route, http
 from apistar.test import TestClient
 
 
+def to_native(obj):  # pragma: nocover  (Some cases only included for completeness)
+    if isinstance(obj, dict):
+        return {key: to_native(value) for key, value in obj.items()}
+    elif isinstance(obj, list):
+        return [to_native(value) for value in obj]
+    elif hasattr(obj, 'read'):
+        content = obj.read()
+        if isinstance(content, bytes):
+            content = content.decode('utf-8')
+        return content
+    elif hasattr(obj, '__dict__'):
+        return str(obj)
+    return obj
+
+
 def get_method(method: http.Method) -> http.Response:
     return http.Response({'method': method})
 
@@ -51,11 +66,11 @@ def get_body(body: http.Body) -> http.Response:
 
 
 def get_data(data: http.RequestData) -> http.Response:
-    return http.Response({'data': data})
+    return http.Response({'data': to_native(data)})
 
 
 def get_field(field: http.RequestField) -> http.Response:
-    return http.Response({'field': field})
+    return http.Response({'field': to_native(field)})
 
 
 def get_headers(headers: http.Headers) -> http.Response:
@@ -189,17 +204,17 @@ def test_body():
 
 def test_data():
     response = client.post('http://example.com/data/', json={"hello": 123})
-    assert response.json() == {'data': {"hello": 123}}
+    assert response.json() == {'data': {'hello': 123}}
 
     response = client.post('http://example.com/data/')
     assert response.json() == {'data': None}
 
     response = client.post('http://example.com/data/', data={'abc': 123})
-    assert response.json() == {'data': {'abc': ['123']}}
+    assert response.json() == {'data': {'abc': '123'}}
 
     csv_file = ('report.csv', '1,2,3\n4,5,6\n')
     response = client.post('http://example.com/data/', files={'file': csv_file})
-    assert response.json() == {'data': {'file': [['1,2,3\n', '4,5,6\n']]}}
+    assert response.json() == {'data': {'file': '1,2,3\n4,5,6\n'}}
 
     response = client.post('http://example.com/data/', headers={'content-type': 'unknown'})
     assert response.status_code == 415
@@ -214,7 +229,7 @@ def test_field():
 
     csv_file = ('report.csv', '1,2,3\n4,5,6\n')
     response = client.post('http://example.com/field/', files={'field': csv_file})
-    assert response.json() == {'field': ['1,2,3\n', '4,5,6\n']}
+    assert response.json() == {'field': '1,2,3\n4,5,6\n'}
 
 
 def test_headers():

--- a/tests/test_routing.py
+++ b/tests/test_routing.py
@@ -169,7 +169,7 @@ def test_invalid_integer():
 
 
 def test_misconfigured_route():
-    def set_type(var: set):  # pragma: nocover
+    def set_type(var: set):  # pragma: nocover  (We never actually call this handler)
         pass
 
     with pytest.raises(exceptions.ConfigurationError):

